oidc-api/src/main/java/uk/gov/di/authentication/oidc/services/UserInfoService.java

```diff
@@ -66,28 +69,69 @@ public String calculateSubjectForAudit(AccessTokenInfo accessTokenInfo) {
 
     public UserInfo populateUserInfo(AccessTokenInfo accessTokenInfo) {
         LOG.info("Populating UserInfo");
-        var userInfo = new UserInfo(new Subject(accessTokenInfo.getSubject()));
+        UserInfo userInfo = new UserInfo(new Subject(accessTokenInfo.getSubject()));
         if (accessTokenInfo.getScopes().contains(CustomScopeValue.DOC_CHECKING_APP.getValue())) {
             return populateDocAppUserInfo(accessTokenInfo, userInfo);
         }
-        var userProfile =
-                authenticationService.getUserProfileFromSubject(
-                        accessTokenInfo.getAccessTokenStore().getInternalSubjectId());
 
-        if (accessTokenInfo.getScopes().contains(OIDCScopeValue.EMAIL.getValue())) {
-            userInfo.setEmailAddress(userProfile.getEmail());
-            userInfo.setEmailVerified(userProfile.isEmailVerified());
-        }
-        if (accessTokenInfo.getScopes().contains(OIDCScopeValue.PHONE.getValue())) {
-            userInfo.setPhoneNumber(userProfile.getPhoneNumber());
-            userInfo.setPhoneNumberVerified(userProfile.isPhoneNumberVerified());
-        }
-        if (accessTokenInfo.getScopes().contains(CustomScopeValue.GOVUK_ACCOUNT.getValue())) {
-            userInfo.setClaim("legacy_subject_id", userProfile.getLegacySubjectID());
-        }
-        if (accessTokenInfo.getScopes().contains(CustomScopeValue.ACCOUNT_MANAGEMENT.getValue())) {
-            userInfo.setClaim("public_subject_id", userProfile.getPublicSubjectID());
+        if (configurationService.isAuthOrchSplitEnabled()) {
+            UserInfo tmpUserInfo = null;
+            try {
+                var userInfoString =
+                        userInfoStorageService
+                                .getAuthenticationUserInfoData(
+                                        accessTokenInfo
+                                                .getAccessTokenStore()
+                                                .getInternalSubjectId())
+                                .get()
+                                .getUserInfo();
+                tmpUserInfo = new UserInfo(JSONObjectUtils.parse(userInfoString));
+            } catch (Exception e) {
+                var error = e.getMessage();
+            }
+
+            if (accessTokenInfo.getScopes().contains(OIDCScopeValue.EMAIL.getValue())) {
+                userInfo.setEmailAddress(tmpUserInfo.getEmailAddress());
+                userInfo.setEmailVerified(tmpUserInfo.getEmailVerified());
+            }
+            if (accessTokenInfo.getScopes().contains(OIDCScopeValue.PHONE.getValue())) {
+                userInfo.setPhoneNumber(tmpUserInfo.getPhoneNumber());
+                userInfo.setPhoneNumberVerified(tmpUserInfo.getPhoneNumberVerified());
+            }
```

Comment:
Should this commit and the next commit get squashed into one? They seem like logically part of the same thing. Benefits: no commenting and uncommenting in the history, unit tests appear with implementation, and easier for people to review if they review commit by commit like I do 😄